src/app/services/configuration-control.service.ts

```diff
@@ -538,4 +538,14 @@ export class ConfigurationControlService {
     return finalConfiguration;
   }
 
+  getValidConfig(config: any) {
+    // remove readonly property form the local configuration copy
+    Object.keys(config).forEach(key => {
+      if (config[key]?.readonly) {
+        delete config[key];
+      }
+    })
```

Comment:
Fine, but may be  return filtered

Reply:
config is an object of properties objects and in there we need to remove the readonly key property, delete is used to remove that property on the same object and returning that object with remaining data.